lib/trino-orc/src/main/java/io/trino/orc/reader/ColumnReaders.java

```diff
@@ -75,6 +76,9 @@ public static ColumnReader createColumnReader(
             case DATE:
                 return new LongColumnReader(type, column, memoryContext.newLocalMemoryContext(ColumnReaders.class.getSimpleName()));
             case FLOAT:
+                if (type == DoubleType.DOUBLE && !column.getAttributes().containsKey("iceberg.id")) {
+                    throw invalidStreamType(column, type);
```

Comment:
Why fail?

- Iceberg has "original files" which won't have `iceberg.id` attribute and it should work for them too
- Hive connector very like wants to have this schema evolution capability as well (cc @Praveen2112 who was asking me about how to do this)

Reply:
Instead of handling in the reader can we handle them in `ColumnAdaption` in case of ORC. In case if we want schema evolution capability like map a Float column to Varchar or someother type - Won't it make the ColumnReader impl too verbose

Reply:
We could use `ColumnAdaptation`, but some numeric column readers already support converting the values in this way.